### DISCUSSION TITLE
Query indexed object property

### DISCUSSION THREAD

**Discussion by marcuslindblom** [ravendb/ravendb#19533] [Q&A] [answered]
I have a field indexed like this `origin: { id: '1234...', collection: 'products' }`. How can I query `origin.id` here?
I tried something like this but it did not work.
```js
const siblings = await session.query<Product>({ indexName: 'indexname' })
  .whereEquals('origin.id', p?.origin?.id)
  .all();
```

**Comment by maciejaszyk** [comment, marked as answer]
Hi,
you're basically indexing the object as:
a) Lucene: JSON string
b) Corax: https://ravendb.net/docs/article-page/6.2/csharp/indexes/search-engine/corax#handling-of-complex-json-objects
You can simply determine what terms do you have under `origin` field by looking at the index's terms page.

You should create fields with the values of `origin` field:
```csharp
from doc in docs.Collection
select new
{
      origin_id = doc.origin.id,
      [...]
}
```
